cwms-data-api/src/main/java/cwms/cda/data/dao/ForecastInstanceDao.java

```diff
@@ -1,37 +1,321 @@
+                        .withOfficeId(officeId)
+                        .withTimeSeriesIds(timeSeriesIdentifiers)
+                        .withSourceEntityId(resultSet.getString(6))
+                        .withLocationId(resultSet.getString(12))
+                        .build())
+                .withDateTime(forecastDate)
+                .withIssueDateTime(issueDate)
+                .withMaxAge(resultSet.getInt(9))
+                .withFileDataUrl(url)
+                .withFileData(fileData)
+                .withFilename(fileName)
+                .withFileMediaType(mediaType)
+                .withNotes(resultSet.getString(11))
+                .withFirstDateTime(firstDateTime)
+                .withLastDateTime(lastDateTime)
+                .withMetadata(forecastInfo)
+                .build();
+    }
+
+    public ForecastInstance getForecastInstance(int byteLimit, ReplaceUtils.OperatorBuilder urlBuilder,
+            String office, String name, String designator,
+            Instant forecastDate, Instant issueDate) {
+        String query = INSTANCE_QUERY + GET_ONE_CONDITIONS;
+        return connectionResult(dsl, c -> {
+            try (PreparedStatement preparedStatement = c.prepareStatement(query)) {
+                preparedStatement.setString(1, office);
+                preparedStatement.setString(2, name);
+                preparedStatement.setString(3, designator);
+                preparedStatement.setLong(4, forecastDate.toEpochMilli());
+                preparedStatement.setLong(5, issueDate.toEpochMilli());
+                try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                    if (resultSet.next()) {
+                        return map(byteLimit, urlBuilder, resultSet);
+                    } else {
+                        String message = format("Could not find forecast instance for " +
+                                        "office id: %s, spec id: %s, designator: %s, forecast date: %s, issue date: %s",
+                                office, name, designator, forecastDate, issueDate);
+                        throw new NotFoundException(message);
+                    }
+                }
+            }
+        });
     }
 
     public void update(ForecastInstance forecastInst) {
-        throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+        String officeId = forecastInst.getSpec().getOfficeId();
+        String specId = forecastInst.getSpec().getSpecId();
+        String designator = forecastInst.getSpec().getDesignator();
+        Instant forecastDate = forecastInst.getDateTime();
+        Instant issueDate = forecastInst.getIssueDateTime();
+        //Will throw a NotFoundException if instance doesn't exist
+        ReplaceUtils.OperatorBuilder noopUrlBuilder = new ReplaceUtils.OperatorBuilder().withTemplate("")
+                .withOperatorKey("{noop}");
+        getForecastInstance(0, noopUrlBuilder, officeId, specId, designator, forecastDate, issueDate);
```

Comment:
This is weird to me but I think I get it.  I would have put more thought into OperatorBuilder if I knew it was going to be part of our api's.

Reply:
yeah, I was just so relieved to get my giant query going, that I didn't spend time on the replaceutils stuff